### PR TITLE
[DT-1059] Implement billing aidl binder getPurchases

### PR DESCRIPTION
**Pull request by pedroagostinhoaptoide**
**What does this PR do?**

   - Implements the getPurchases method in the aidl binder, that is responsible for fetching the purchases related to a merchant and of a certain type.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] See commit.

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [DT-1059](https://aptoide.atlassian.net/browse/DT-1059)

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-1059](https://aptoide.atlassian.net/browse/DT-1059)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[DT-1059]: https://aptoide.atlassian.net/browse/DT-1059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-1059]: https://aptoide.atlassian.net/browse/DT-1059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: payments/base/arch/src/main/java/com/appcoins/payments/arch/PurchaseInfoData.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  val productName: String,
+  val state: PurchaseState,
+  val autoRenewing: Boolean,
+  val renewal: Date?,
+  val packageName: String,
+  val signatureValue: String,
+  val signatureMessage: String,
+)
+
+@Keep
```
Comment (ricardo-frango) [inline payments/base/arch/src/main/java/com/appcoins/payments/arch/PurchaseInfoData.kt:17]: do we need the keep here?

File: payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt [hunk 2]
```diff
@@ -33,4 +35,13 @@ internal class BillingErrorMapperImpl @Inject constructor() : BillingErrorMapper
       else -> BillingSdkConstants.ResultCode.RESULT_ERROR
     }
   }
+
+  override fun mapPurchasesError(exception: Throwable): Int {
+    return when (exception) {
+      is HttpException -> BillingSdkConstants.ResultCode.RESULT_ERROR
+      is IOException -> BillingSdkConstants.ResultCode.RESULT_SERVICE_UNAVAILABLE
```
Comment (ricardo-frango) [inline payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt:42]: Should be UnknownHostException

File: payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt [hunk 3]
```diff
@@ -33,4 +35,13 @@ internal class BillingErrorMapperImpl @Inject constructor() : BillingErrorMapper
       else -> BillingSdkConstants.ResultCode.RESULT_ERROR
     }
   }
+
+  override fun mapPurchasesError(exception: Throwable): Int {
+    return when (exception) {
+      is HttpException -> BillingSdkConstants.ResultCode.RESULT_ERROR
```
Comment (ricardo-frango) [inline payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt:41]: This case could be removed

**Review by ricardo-frango — CHANGES_REQUESTED** [review]

File: payments/base/arch/src/main/java/com/appcoins/payments/arch/PurchaseInfoData.kt (hunk 1 above)
Comment (pedroagostinhoaptoide) [reply payments/base/arch/src/main/java/com/appcoins/payments/arch/PurchaseInfoData.kt]: Removed it

File: payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt (hunk 2 above)
Comment (pedroagostinhoaptoide) [reply payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt]: Fixed

File: payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt (hunk 3 above)
Comment (pedroagostinhoaptoide) [reply payments/billing-sdk/src/main/java/com/appcoins/billing/sdk/BillingErrorMapper.kt]: Fixed

**Review by ricardo-frango — APPROVED** [review]

**Pull request merged**